logstreams/src/main/java/io/zeebe/logstreams/storage/atomix/AtomixRaftServer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.logstreams.storage.atomix;
+
+import io.atomix.protocols.raft.partition.impl.RaftPartitionServer;
+import io.atomix.protocols.raft.storage.log.RaftLogReader;
+import io.atomix.protocols.raft.zeebe.ZeebeLogAppender;
+import io.atomix.storage.journal.JournalReader.Mode;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public class AtomixRaftServer
+    implements AtomixLogCompactor, AtomixReaderFactory, AtomixAppenderSupplier {
+  private final RaftPartitionServer server;
+
+  public AtomixRaftServer(final RaftPartitionServer server) {
+    this.server = server;
+  }
+
+  @Override
+  public Optional<ZeebeLogAppender> getAppender() {
+    return server.getAppender();
+  }
+
+  @Override
+  public CompletableFuture<Void> compact(final long index) {
+    server.setCompactablePosition(index, 0); // todo: remove term
```

Comment:
This has to be done in atomix then? (removing the term)

Reply:
Follow up issue [atomix#102](https://github.com/zeebe-io/atomix/issues/102)